opendatahub-io/odh-dashboard
# Merge feature branch f/storage-class

Closes: [RHOAIENG-2872](https://issues.redhat.com/browse/RHOAIENG-2872)

## Description


Adds ability to set a storage class via Dashboard CR value `storageClassName` when no default storage class is set on the cluster

## How Has This Been Tested?



Test with and without a default storage class. 

When testing without a default storage class:
1. Set dashboard CR value `storageClassName` to a given storage class (you can create a new one if neccessary).
2. From the project detail page, create a "Cluster storage"
3. Verify the correct storage class is assigned to the PVC via the openshift console.
4. Create a Workbench, create a new storage class in the process, verify the PVC as in step 3.
5. Edit the above work bench, in the modal that pops up, create a new PVC again and verify as in step 3.

Test with a default storage class:
1. Set a default storage class on the cluster
2. Follow test instructions exactly as above, but verify at each step that the storage class is set to the default, not the CR value. 

## Test Impact


Relevant tests have been updated to work with new features.

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
